### DISCUSSION TITLE
changing pages directory through middleware

### DISCUSSION THREAD

**Discussion by PhenomEY** [nuxt/nuxt#17324] [Questions]
Hello,

is it possible to change the pages directory through middleware?

i have 2 pages directories:

main
second

each one contains sub-pages

if a user is permitted he will be redirected to the pages in /main
else he is gonna see the pages from /second

the tricky part here is that both should happen under the url / 

In nuxt2 this was possible using the router module but i cant get this to work in nuxt3.
I think middleware is the way to go in nuxt3 for that?

**Comment by alpeshvmodi** [comment]
I am also curious to see solution for this.
